modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/FilePerformanceStatisticsWalker.java

```diff
@@ -0,0 +1,286 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.performancestatistics;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.ignite.internal.processors.cache.persistence.file.FileIO;
+import org.apache.ignite.internal.processors.cache.persistence.file.RandomAccessFileIOFactory;
+import org.apache.ignite.internal.processors.cache.query.GridCacheQueryType;
+import org.apache.ignite.internal.processors.performancestatistics.IgnitePerformanceStatistics.CacheOperationType;
+import org.apache.ignite.internal.util.GridIntList;
+import org.apache.ignite.internal.util.GridUnsafe;
+import org.apache.ignite.lang.IgniteUuid;
+
+import static java.nio.ByteBuffer.allocateDirect;
+import static java.nio.ByteOrder.nativeOrder;
+import static org.apache.ignite.internal.processors.performancestatistics.FilePerformanceStatistics.readIgniteUuid;
+import static org.apache.ignite.internal.processors.performancestatistics.FilePerformanceStatistics.readUuid;
+
+/**
+ * Walker over the performance statistics file.
+ *
+ * @see FilePerformanceStatistics
+ */
+public class FilePerformanceStatisticsWalker {
+    /** File read buffer size. */
+    private static final int READ_BUFFER_SIZE = 8 * 1024 * 1024;
+
+    /** IO factory. */
+    private static final RandomAccessFileIOFactory ioFactory = new RandomAccessFileIOFactory();
+
+    /**
+     * Walks over performance statistics file.
+     *
+     * @param file Performance statistics file.
+     * @param handlers Handlers to process deserialized operation.
+     */
+    public static void walkFile(Path file, IgnitePerformanceStatistics... handlers) throws IOException {
+        ByteBuffer buf = allocateDirect(READ_BUFFER_SIZE).order(nativeOrder());
+
+        try (
+            FileIO io = ioFactory.create(file.toFile());
+            PerformanceStatisticsDeserializer des = new PerformanceStatisticsDeserializer(handlers)
+        ) {
+            while (true) {
+                int read = io.read(buf);
+
+                buf.flip();
+
+                if (read <= 0)
+                    break;
+
+                while (true) {
```

Comment:
Let's rewrite it to the `while (des.deserialize(buf));`

Reply:
Fixed.